specification/dimensions/skuid.md

```diff
@@ -0,0 +1,91 @@
+## Spec Content
+
+### SKUID
+
+A SKU ID is an identifier assigned to the SKU that’s associated with a charge. SKU ID can be referenced on a price list published by a provider to look up detailed information about the SKU including the different unit prices associated with the SKU.
+
+The SKUId column MUST be present in the billing data. This column MUST be of type String. The SKUId MUST NOT be null when [Charge Type](#ChargeType) is ‘Purchase’ or ‘Usage’. The SKUId SHOULD NOT be null when [Charge Type](#ChargeType) is ‘Adjustment’ or [Adjustment Type](#AdjustmentType) is ‘Refund’.
```

Comment:
```suggestion
The SKUId column MUST be present in the billing data. This column MUST be of type String. The SKUId MUST NOT be null when [Charge Type](#ChargeType) is ‘Purchase’ or ‘Usage’. The SKUId SHOULD NOT be null when [Adjustment Type](#AdjustmentType) is ‘Refund’. The SKUId MUST be null when [Adjustment Type](#AdjustmentType) not ‘Refund’.
```
I'm not sure I fully understand what the values listed for Adjustment type are tbh. But ultimately, what we want to suggest here is that if you're providing an adjustment where its related to a particular SKU then the SKU ID needs to be specified. If it's more of an ad-hoc adjustment, then it does not need to be set.

Reply:
@udam-f2 This section was directly lifted from your v1 draft... I'm sure @arrarama is happy to make the change per your suggestion.

Reply:
Upon discussion, the suggestion works with the following revision:

> The SKUId column MUST be present in the billing data. This column MUST be of type String. The SKUId MUST NOT be null when [Charge Type](#ChargeType) is ‘Purchase’ or ‘Usage’. The SKUId SHOULD NOT be null when [Adjustment Type](#AdjustmentType) is ‘Refund’.

The stricture around forcing null for Adjustment Type != Refund feels too strong per Mike Fuller.

Reply:
I think other groups have gone with the approach of saying when somehting MUST be or SHOULD BE null and also adding when it MUST NOT be null. This wasn't what we did in 0.5 so its on my list of inconsistencies to bring up with the group at the end.

What I meant above on AdjustmentType was I'm not sure I understand the values VS2 is proposing on AdjustmentType but the idea of the values per my understanding was that if it's a refund, then it will have SKU and other relevant info (unit rates, quantities etc.). If it doesn't have those, then it cant be a refund and is instead an adjustment. Now that 2-3 other adjustment types are defined, im not sure where the boundary lies. Would be good to follow up with VS2 since without some guidance, providers will have to chose when and when not to put the value here. (I understand this is even inconsistent with my own suggestion above :))